applications/MeshingApplication/custom_utilities/tetrahedra10_mesh_converter_utility.h

```diff
@@ -0,0 +1,356 @@
+// KRATOS  __  __ _____ ____  _   _ ___ _   _  ____
+//        |  \/  | ____/ ___|| | | |_ _| \ | |/ ___|
+//        | |\/| |  _| \___ \| |_| || ||  \| | |  _
+//        | |  | | |___ ___) |  _  || || |\  | |_| |
+//        |_|  |_|_____|____/|_| |_|___|_| \_|\____| APPLICATION
+//
+//  License:		 BSD License
+//                                       Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ariadna Cortés
+//
+
+#if !defined(KRATOS_TETRAHEDRA10_MESH_CONVERTER_UTILITY)
+#define  KRATOS_TETRAHEDRA10_MESH_CONVERTER_UTILITY
+
+// System includes
+
+/* Project includes */
+#include "custom_utilities/local_refine_tetrahedra_mesh.hpp"
+#include "geometries/tetrahedra_3d_10.h"
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+///@}
+///@name Type Definitions
+///@{
+///@}
+///@name  Enum's
+///@{
+///@}
+///@name  Functions
+///@{
+///@}
+///@name Kratos Classes
+///@{
+class Tetrahedra10MeshConverter : public LocalRefineTetrahedraMesh
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+    ///@}
+
+    typedef Node<3> NodeType;
+    typedef Geometry<NodeType> GeometryType;
+    typedef GeometryType::Pointer GeometryPtrType;
+
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructors
+    Tetrahedra10MeshConverter(ModelPart& model_part) : LocalRefineTetrahedraMesh(model_part)
+    {
+
+    }
+
+    /// Destructor
+    ~Tetrahedra10MeshConverter() 
+    = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+        
+    /**
+    * Replaces the geometry of the mesh of Tetrahedra3D4 elements. Resulting is a mesh of Tetrahedra3D10 created
+    * by adding intermediate nodes to each Tetrahedra3D4, with same ids as the originals. Does the same for conditions,
+    * replacing the Triangle3D3 by Triangle3D6
+    * @param refine_on_reference: Boolean that defines if refine or not the mesh according to the reference
+    * @param interpolate_internal_variables: Boolean that defines if to interpolate or not the internal variables
+    */
+    void LocalConvertTetrahedra10Mesh(bool refine_on_reference, bool interpolate_internal_variables) {
+            for (auto element : mModelPart.Elements()) element.SetValue(SPLIT_ELEMENT,true);
+            for (auto condition : mModelPart.Conditions()) condition.SetValue(SPLIT_ELEMENT,true);
+            LocalRefineMesh(refine_on_reference, interpolate_internal_variables);
```

Comment:
You may store the `interpolate_internal_variables` as a member variable to avoid passing it to all methods.

Reply:
This would not be very practical, since the class is overriding the base class methods (which pass this variable). If we store it as a member variable and declare new methods that dont need this passing, this methods will no longer be overriding the old ones (but overloading?), so the code could not work in the same way it does right now.